salt/metalk8s/kubeadm/init/certs/apiserver.sls

```diff
@@ -0,0 +1,34 @@
+{%- from "metalk8s/map.jinja" import kube_api with context %}
+
+Create kube api private key:
+  x509.private_key_managed:
+    - name: {{ kube_api.cert.public_key }}
+    - mode: 600
+    - makedirs: True
+
+{#- Add FQDN of the server #}
+{%- do kube_api.cert.subjectAltName.get('DNS').insert(0, grains['fqdn']) %}
+
+{#- TODO: Not use all IPs and some logic to get service IP #}
+{%- do kube_api.cert.subjectAltName.get('IP').append(
+         salt['netaddress.list_cidr_ips'](kube_api.service_ip_range)[1]) %}
+{%- for ip in grains['ipv4'] %}
+  {%- if ip != '127.0.0.1' %}
+    {%- do kube_api.cert.subjectAltName.get('IP').append(ip) %}
+  {%- endif %}
+{%- endfor %}
+
+{#- Convert dictionary to string #}
+{%- do kube_api.cert.update(
+         {'subjectAltName': salt['metalk8s_certs.compute_subject_alt_name'](
+                              kube_api.cert.subjectAltName)}) %}
+Generate kube api certificate:
+  x509.certificate_managed:
+    {%- for key, value in kube_api.cert.items() %}
+    - {{ key }}: {{ value }}
+    {%- endfor %}
+    {#- If ca_server is empty then use local server as CA #}
+    - ca_server: {{ (salt['mine.get']('*', 'ca_server').keys() or [''])[0] }}
+    - makedirs: True
+    - require:
```

Comment:
Does this imply the cert will be regenerated when the key changes, but only then? I.e. we don't want the cert to be regenerated when this state is applied again, unless the cert doesn't match the given key (i.e. the key was changed, so the cert must be changed), or when the cert expired.

Reply:
The salt states already check the cert so it's automagicaly handle that only generate a new cert if the key changed or something else changed in the cert like CN of what ever. If we call several time the state that will only generate it one time.

I just need to test if that re-generate a certificate if the certificate expired (will check)

Reply:
Ok btw we have this option in the salt states who default to 90.
```
days_remaining
: 90
The minimum number of days remaining when the certificate should be recreated. A value of 0 disables automatic renewal.
```

Reply:
OK, cool